### ISSUE TITLE
WarnServ should support !warn expires

### ISSUE THREAD

**Issue by shymega** [vktec-irc/warnserv#9]
When admins `!warn` someone, there should be an optional expiry
argument.

On several occasions, I have warned people, and forgotten to reset the
warnings.

With an expiry argument, we can get WarnServ to automagically expire warnings.

**Comment by vktec** [comment]
Good idea. The implementation of this would probably involve multiple threads (one to listen for messages, one to handle timeouts), but that can all be done, and is actually rather easy with `srfi-18` and the `mailbox` egg
